velero-plugins/pvc/vmfr_restore.go

```diff
@@ -51,25 +52,64 @@ func (p *VMFRRestorePlugin) Execute(input *velero.RestoreItemActionExecuteInput)
 		return nil, fmt.Errorf("[vmfr-pvc-restore] error unmarshaling PVC: %v", err)
 	}
 
-	// Generate unique PVC name using GenerateName to avoid collisions.
-	// The Kubernetes API server will automatically truncate long prefixes (>58 chars)
-	// and append a 5-character random suffix to ensure uniqueness.
+	// Generate unique PVC name to avoid collisions when restoring multiple backups.
+	// Pattern: {backup-name}-{original-pvc-name}-{5-char-random-suffix}
+	// This ensures uniqueness even if names get truncated.
 	originalName := pvc.Name
 	backupName := input.Restore.Spec.BackupName
-	generateNamePrefix := fmt.Sprintf("%s-%s-", backupName, originalName)
 
-	p.Log.Infof("[vmfr-pvc-restore] Setting GenerateName for PVC from %s to %s for VMFR multi-backup restore", originalName, generateNamePrefix)
+	// Generate 5-character random suffix using Kubernetes utilities (same as GenerateName)
+	// This uses lowercase alphanumeric characters, consistent with K8s API server behavior
+	suffix := rand.String(5)
 
-	// Use GenerateName instead of Name to let Kubernetes generate unique suffix
-	pvc.GenerateName = generateNamePrefix
-	pvc.Name = ""
+	// Kubernetes resource names must be <= 253 characters (DNS-1123 subdomain)
+	// Smart truncation: keep backup-name prefix, truncate pvc-name if needed, always keep suffix
+	maxLen := 253
+	reservedLen := len(backupName) + 1 + 1 + 5 // backup-name + "-" + "-" + 5-digit-suffix
```

Comment:
generateName retry is GA.
https://github.com/kubernetes/kubernetes/pull/127093

This has no retry is there?

I would vote in favor of using generateName with a shorter prefix, or maybe just fully generatename..
you can filter backupname using label.

`Leave room for at least 1 char of PVC name` is not particularly reassuring to me that a backup would contain less than 26 PVCs.. so even if you have retry.. 27PVC will fail

Reply:
No, we can not use `generateName` here. Velero will complain that the name can not be empty and will fail restore. We need to generate name ourselves + we need to have at least 1 pvc with original name, otherwise velero will fail and we can not solve this in plugin as it's main velero logic.